---
title: "Class 13: Genome Informatics 1""
output: github_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## MXL genotype analysis 
How many G|G genotypes? 

```{r}
mxl <- read.csv("373531-SampleGenotypes-Homo_sapiens_Variation_Sample_rs8067378.csv")
head(mxl)
```
```{r}
genotypes <- mxl$Genotype..forward.strand.
table(genotypes)

```
```{r}
table(genotypes)/nrow(mxl)*100

```

##FASTQ Quality Scores
```{r}
#install.packages("seqinr")
#install.packages("gtools")
library(seqinr)
library(gtools)
chars <- s2c("DDDDCDEDCDDDDBBDDDCC@")
chars
phred <- asc(chars) - 33
phred
```


##Population Scale Analysis
```{r}
expr <- read.table("rs8067378_ENSG00000172057.6.txt" )
head(expr)

```
```{r}
table(expr$geno)
```
```{r}
inds.gg <- expr$geno == "G/G"
expr[inds.gg,"exp"]
summary(expr[inds.gg,"exp"])
```

```{r}

inds.ag <- expr$geno == "A/G"
expr[inds.ag,"exp"]
summary(expr[inds.ag,"exp"])
```
```{r}

inds.aa <- expr$geno == "A/A"
expr[inds.aa,"exp"]
summary(expr[inds.aa,"exp"])
```

how to show this data
```{r}
#how to run boxplot across genotypes?
boxplot(exp ~ geno, data = expr, notch = TRUE)
```

